app/javascript/components/bookmarks/BookmarkManager.jsx

```diff
@@ -0,0 +1,286 @@
+    const saveProps = [currentBookmark]
+    const saveFunc = bookmarkSaved ? updateBookmark : createBookmark
+    if (bookmarkSaved) {
+      saveProps.unshift(getBookmarkId(currentBookmark))
+    }
+    try {
+      const bookmark = await saveFunc(...saveProps)
+      enableSaveButton(true)
+      setShowError(false)
+      resetForm()
+      updateBookmarkList(bookmark)
+    } catch (error) {
+      enableSaveButton(true)
+      handleErrorContent(error)
+    }
+  }
+
+  /** delete a bookmark */
+  async function handleDeleteBookmark(e) {
+    e.preventDefault()
+    setDeleteDisabled(true)
+    const toDelete = getBookmarkId(currentBookmark)
+    if (!toDelete) {
+      setDeleteDisabled(false)
+      return false
+    } else {
+      try {
+        await deleteBookmark(toDelete)
+        setDeleteDisabled(false)
+        setShowError(false)
+        removeBookmarkFromList(toDelete)
+      } catch (error) {
+        setDeleteDisabled(false)
+        handleErrorContent(error)
+      }
+    }
+  }
+
+  const [serverBookmarks, setServerBookmarks] = useState([])
+  const [serverBookmarksLoaded, setServerBookmarksLoaded] = useState(false)
+  const [showBookmarksModal, setShowBookmarksModal] = useState(false)
+
+  const toggleBookmarkModal = () => {
+    setShowBookmarksModal(!showBookmarksModal)
+  }
+
+  /** load all user bookmarks from server */
+  async function loadServerBookmarks() {
+    toggleBookmarkModal()
+    if (!serverBookmarksLoaded) {
+      try {
+        const serverUserBookmarks = await fetchBookmarks()
+        setServerBookmarks(serverUserBookmarks)
+        setServerBookmarksLoaded(true)
+      } catch (error) {
+        setShowBookmarksModal(false)
+        setServerBookmarks([])
+        setServerBookmarksLoaded(false)
+      }
+    }
+  }
+
+  const loginPopover = <Popover data-analytics-name='login-bookmark-popover' id='login-bookmark-popover'>
+    You must sign in to bookmark this view
+  </Popover>
+
+  const bookmarkForm = <Popover data-analytics-name='bookmark-form-popover' id='bookmark-form-popover'>
+    <form id='bookmark-form' onSubmit={handleSaveBookmark}>
+      { ErrorComponent }
+      <div className="form-group">
+        <label htmlFor='bookmark-name'>Bookmark name</label>&nbsp;
+        <br/>
+        <input className="form-control"
+               type="text"
+               id='bookmark-name'
+               value={currentBookmark.name}
+               onChange={handleFormUpdate}
+        />
+      </div>
+      <div className="form-group">
+      <label htmlFor='bookmark-description'>Description</label><br/>
+        <textarea className="form-control"
+               id='bookmark-description'
+               value={currentBookmark.description}
+               onChange={handleFormUpdate}
+        />
+      </div>
+      <div className="form-group">
+      <button
+          type="button"
+          className="btn btn-primary"
+          aria-label="Save"
+          data-analytics-name="bookmark-submit"
```

Comment:
Aligning this name more with the UI would make analytics queries more intuitive.  

Also, there's a local word-order inconsistency in these analytics names.  It's currently (pending above suggestion) `bookmark-save`, `bookmark-delete`, and `manage-bookmarks`.  It should be either `bookmark-save`, `bookmark-delete`, and `bookmarks-manage`, or `save-bookmark`, `delete-bookmark`, and `manage-bookmarks`.  I'd slightly lean towards the latter, as verb-noun is more aligned with the UI and also more intuitive for analytics queries.

```suggestion
          data-analytics-name="save-bookmark"
```